leasing/report/utils.py

```diff
@@ -3,3 +3,35 @@ def dictfetchall(cursor):
     """Return all rows from a cursor as a dict"""
     columns = [col[0] for col in cursor.description]
     return [dict(zip(columns, row)) for row in cursor.fetchall()]
+
+
+# Gaps in billing periods helpers
+
+
+def set_active_rent_period(row, today, start, end):
+    """
+    Set the active rent period for the lease to be compared with the billing periods.
+    """
+    start = row["start_date"]
+    if not row["end_date"] or row["end_date"] > today:
+        end = today
+    else:
+        end = row["end_date"]
+
+    return start, end
+
+
+def calculate_invoice_billing_period_days(start_date, end_date, today):
+    """
+    Calculate the billing period days for the invoice.
+    Returns None if there are missing dates.
```

Comment:
Does this `Returns None if there are missing dates.` mean that if start_date or end_date is missing?
If that is the case only, it could simplify the logic by inversing the check:
```
if start_date is None or end_date is None:
   return None
```